### DISCUSSION TITLE
Changing facial recognition photo?

### DISCUSSION THREAD

**Discussion by bcsteeve** [immich-app/immich#3357] [General]
Immich decided to pick a TERRIBLE photo of me to represent me as the known face :)

There's a "change feature photo" option, which I assumed would allow me to change this photo, but that doesn't appear to do anything at all.

What is the "feature photo"?  Is there a way to change the photo representing the "face" for a person?

**Comment by bo0tzz** [comment]
That should be the way. Did you get a notification in the top right that the photo was updated? Are there any errors in the browser console or container logs?

**Comment by daniele-athome** [comment]
Remember to refresh the page after doing that, the change is not instantaneous.

**Comment by alextran1502** [comment]
@bcsteeve You will need to run the face detection job again in order for some additional information get written to the database as noted in the release note when this feature was implemented

**Comment by bcsteeve** [comment]
Ok, so I ran the face detection job and tried again... nothing.
I reloaded the page a dozen times... nothing.

**I tried it in Edge after only trying in Chrome**... it worked (after reloading the page).

BUT... only once.   I can't change the same photo again.  I tried clearly browsing history and trying again. Tried running the job again (that takes a long time), but still can't change it again.

I'm not sure it works as it is intended.

**Reply by alextran1502** [reply]
@bcsteeve did you try to run for all faces?

**Reply by bcsteeve** [reply]
Hmm... maybe not.  That's necessary?

Each time I want to change one?  Because it destroys all my face matching work (naming, but also merging?).  I don't mind doing it once, but...

**Reply by alextran1502** [reply]
@bcsteeve you only need to do it once so that the faces coordinated get written to the database. 

When we first roll out facial recognition, it was an oversight that we didn't think about this use case so the faces coordinate didn't get register in the database. Then when we added this changing feature photo of a person, we required detect faces job needed to be run again on all faces for it to work

**Reply by bcsteeve** [reply]
Gotcha.  I saw the requirement to run the job, but I guess I missed the "all" requirement, although that does seem kind of obvious now that you mention it lol.
